awscrt/auth.py

```diff
@@ -18,21 +18,31 @@
 from concurrent.futures import Future
 
 
-class Credentials(object):
+class Credentials(NativeResource):
     """
     Credentials are the public/private data needed to sign an authenticated AWS request.
     """
 
-    __slots__ = ('access_key_id', 'secret_access_key', 'session_token')
+    __slots__ = ()
```

Comment:
uuuh how about `_binding`?

Reply:
It's in to `NativeResource.__slots__`
Slots are added by each step in the class hierarchy